src/js/item.ts

```diff
@@ -64,6 +70,31 @@ export function extractItems(
   return [];
 }
 
+
+// TODO not the best place for this method, should rearrange and possibly change the callsite where this is used
+export function getCategoriesForProduct(scheduler: IRequestScheduler, productUrl: string): Promise<string> {
+    return scheduler.scheduleToPromise<string>(
+        productUrl,
+        // the result of this transformation is cached; clear cache when debugging & changing this line
+        // the scheduler cache uses localstorage which has a very limited storage size, so do not store full page results
+
+        (evt) => {
+            const productPage = util.parseStringToDOM(evt.target.responseText).documentElement;
+
+            try {
```

Comment:
I like how you've done this without using any user visible strings to find it - e.g. less likely to work for only some sites.

Reply:
s/less likely to work for only some sites/more likely to work for all sites/